Assignment 2.Rmd

```diff
@@ -81,20 +86,107 @@ plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender)
 
 plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender, edge.width=EDGE$count)
 
-````
+```
 
 ## Part II
 
 In Part II your task is to [look up](http://igraph.org/r/) in the igraph documentation and create a graph that sizes the student vertices in terms of the number of disciplinary actions they have recieved, and the teachers in terms of the number of disciplinary actions they have given out.
 
+```{r}
+#correct approach
+Vertex_T <- EDGE %>% group_by(from) %>% summarize(sum(count))
+Vertex_S <- EDGE %>% group_by(to) %>% summarize(sum(count))
+colnames(Vertex_T) <- c("id", "count")
+colnames(Vertex_S) <- c("id", "count")
+Vertex <- bind_rows(Vertex_T, Vertex_S)
+
+Vertex_g <- graph.data.frame(EDGE, directed = TRUE, vertices = Vertex)
+scale_factor <- 2
+plot(Vertex_g,
+     layout=layout.fruchterman.reingold, 
+     vertex.color=VERTEX$gender, 
+     vertex.size=Vertex$count * scale_factor, 
+     margin = -0.3,
+     layout = layout_nicely)
 
+```
+
+```
 
 ## Part III
 
 Now practice with data from our class. Please create a **person-network** with the data set hudk4050-classes.csv. To create this network you will need to create a person-class matrix using the tidyr functions and then create a person-person matrix using `t()`. You will then need to plot a matrix rather than a data frame using igraph.
 
 Once you have done this, also [look up](http://igraph.org/r/) how to generate the following network metrics: betweeness centrality and dregree. **Who is the most central person in the network?**
 
+```{r}
+library(tidyverse)
+class_file <- as.matrix(read.csv("hudk4050-classes.csv", header = TRUE))
+class(class_file)
+```
+
+```{r}
+try_1 <- read.csv("hudk4050-classes.csv", header = TRUE)
+try_1 <- unite(try_1, "Full.Name", First.Name, Last.Name, sep = " ")
+
+#eliminate white space
+test_space <- select(try_1, 2:7)
+test_space <- as.data.frame(apply(test_space,2,function(x)gsub('\\s+', '',x)))
+
+#only name and class name remain
+Name <- select(try_1, 1)
+try_2 <- cbind(Name, test_space)
+try_2 <- gather(try_2, "class", "case", 2:7)
+try_2 <- try_2[, -2]
+
+#try to create a table with students' number of attending each class
+D3 <- select(try_2, Full.Name, case)
+EDGE_1 <- count(D3, Full.Name, case)
+names(EDGE_1) <- c("Name", "Class", "count")
+
+#delete rows with empty cells under class
+EDGE_1 <- EDGE_1[-which(EDGE_1$Class == ""), ]
+
+#spread according to class
+EDGE_2 <- spread(EDGE_1, Class, count)
+EDGE_2[is.na(EDGE_2)] <- 0
+
+#set fullname as row names
+EDGE_2 <- EDGE_2 %>% remove_rownames %>% column_to_rownames(var="Name")
+
+#switch to matrix and transpose
+EDGE_2 <- as.matrix(EDGE_2)
+t_1 <- t(EDGE_2)
+
+#switch to person to person matrix and eliminate diagnoal redundancy 
+t_2 <- EDGE_2 %*% t_1
+diag(t_2) <- 0
+
+#eliminate outlier, optional, just keep graph pretty
+t_3 <- t_2[rownames(t_2) != "ZIMO CHEN", colnames(t_2) != "ZIMO CHEN"]
```

Comment:
No problem to do this here. But next time when doing important projects, think carefully before simply dropping the outliers.